src/utils/mod.rs

```diff
@@ -68,3 +69,14 @@ pub enum Style {
     Warning,
     Error,
 }
+
+pub fn maybe_json(data: Option<String>) -> Option<Map<String, Value>> {
```

Comment:
I think you could do this with https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_parser the good thing about this is that you'll get feedback from Clap when the value is incorrect

Reply:
I looked into this. Looks like the parser ONLY does validation. I want to actually convert the data. Converting will tell me whether it's valid or not so this would be an extra step.

Reply:
You can convert the values with `value_parser` and `FromStr` like this 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=99878a7c31bf79048c92a77c06096969

or this 

https://www.reddit.com/r/learnrust/comments/115ldw2/implement_parser_for_a_custom_type_with_clap/

but I won't hold the PR for it ^^